app/frontend/src/features/auth/signup/Signup.test.tsx

```diff
@@ -327,4 +333,54 @@ describe("Signup", () => {
     mockConsoleError();
     await assertErrorAlert("Permission denied");
   });
+
+  it("sets the email flow state correctly when given a url token", async () => {
+    signupFlowEmailTokenMock.mockResolvedValue({
+      needBasic: false,
+      needAccount: true,
+      needFeedback: true,
+      needVerifyEmail: false,
+      flowToken: "token",
+    });
+    const state: SignupFlowRes.AsObject = {
+      needBasic: false,
+      needAccount: true,
+      needFeedback: true,
+      needVerifyEmail: true,
+      flowToken: "token",
+    };
+    window.localStorage.setItem("auth.flowState", JSON.stringify(state));
+    const wrapper = getHookWrapperWithClient({
+      initialRouterEntries: [`${signupRoute}/fakeEmailToken`],
+    }).wrapper;
+    render(<View />, {
+      wrapper,
+    });
+    expect(await screen.findByLabelText(USERNAME)).toBeVisible();
+    expect(signupFlowEmailTokenMock).toBeCalledWith("fakeEmailToken");
+    const { result } = renderHook(() => useAuthStore(), { wrapper });
+    expect(result.current.authState.flowState?.needVerifyEmail).toBe(false);
```

Comment:
Couldn't we look into `localStorage` and assert the result there instead?

Reply:
I thought about that, but local storage is basically hacked in so thought it's not a great idea, and this works fine